src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/UpdateDocumentById.cs

```diff
@@ -223,8 +219,42 @@ await _sqlAction.DeleteReferencesByDocumentUuid(
                         }
                     }
 
+                    var cascadingUpdateResults = await _sqlAction.CascadeUpdates(
```

Comment:
This looks like it runs for any and all updates? It only needs to cascade on a resource with `updateRequest.ResourceInfo.AllowIdentityUpdates == true` and where `validationResult.ReferentialIdUnchanged == false`, like the checks around lines 99-102.